src/VintageBookshelf.Data/Repository/BookshelfRepository.cs

```diff
@@ -7,5 +10,20 @@ namespace VintageBookshelf.Data.Repository
     public sealed class BookshelfRepository : Repository<Bookshelf>, IBookshelfRepository
     {
         public BookshelfRepository(VintageBookshelfContext db) : base(db) { }
+        public async Task<Bookshelf> GetBookshelfWithProducts(long id)
```

Comment:
Why don't you create a private **Private Task IQueriable<Bookshelf> GetBookshelfWithProducts() => ...** ? Then you can call it inside the other methods.